docs/map-making/how-to-upgrade-legacy-map-features.md

```diff
@@ -35,70 +19,36 @@ All occurrences of the code before the `>>` must be substituted with the code af
 - `="conditionType" count="XOR"` >> `="conditionType" count="1"`
 - `="conditionType" count="-reset-XOR"` >> `="conditionType" count="-reset-1"`
 - `="isTwoHit" count="true"` >> `="hitPoints" count="2"`
-- `="isTwoHit" value="false"` >> `="hitPoints" value="1"`
-- `="isTwoHit" count="-reset-true"` >> `="hitPoints" count="-reset-2"`
 - `="isTwoHit" count="false"` >> `="hitPoints" count="1"`
+- `="isTwoHit" value="true"` >> `="hitPoints" value="2"`
+- `="isTwoHit" value="false"` >> `="hitPoints" count="1"`
+- `="isTwoHit" count="-reset-true"` >> `="hitPoints" count="-reset-2"`
 - `="isTwoHit" count="-reset-false"` >> `="hitPoints" count="-reset-1"`
 - `="occupiedTerrOf"` >> `="originalOwner"`
 - `="victoryCity" count="true"` >> `="victoryCity" count="1"`
-- `="victoryCity" count="-reset-true"` >> `="victoryCity" count="-reset-1"`
 - `="victoryCity" count="false"` >> `="victoryCity" count="0"`
+- `="victoryCity" value="true"` >> `="victoryCity" count="1"`
+- `="victoryCity" value="false"` >> `="victoryCity" count="0"`
+- `="victoryCity" count="-reset-true"` >> `="victoryCity" count="-reset-1"`
 - `="victoryCity" count="-reset-false"` >> `="victoryCity" count="-reset-0"`
-
-
-### **Procedural alternatives:**
-
-For a more defined change, instead of the aforementioned:
-
-- `attatchment` >> `attachment`
-
-you can change:
-
-- `attatchmentList` >> `attachmentList`
-- `<attatchment` >> `<attachment`
-- `attatchment>` >> `attachment>`
-- `="games.strategy.triplea.attatchments` >> `="games.strategy.triplea.attachments`
-
-- `="techAttatchment` >> `="techAttachment`
-- `="unitAttatchment` >> `="unitAttachment`
-- `="territoryAttatchment` >> `="territoryAttachment`
-- `="canalAttatchment` >> `="canalAttachment`
-- `="rulesAttatchment` >> `="rulesAttachment`
-- `="playerAttatchment` >> `="playerAttachment`
+- `<option name="isParatroop"` >> `<option name="isAirTransportable"`
+- `<option name="isMechanized"` >> `<optiona name="isLandTransportable"`
```

Comment:
All that I said in my previous comment applies here too, changing "isParatroop" with "isMechanized" and "isAirTransportable" with "isLandTransportable".

Reply:
Ditto, I am pretty confident this is the correct update to be done.


commit 67caac77e8b3da9ac759c0f2283806b89e68f70d
Author: Ron <Ron@RMURHAMM-W10E.fios-router.home>
Date:   Thu Dec 7 17:20:08 2017 -0600

    Add isLandTransportable property and some refactoring

+  // isInfantry is DEPRECATED, use isLandTransportable

Reply:
My second comment at my previous comment still applies here too, the same way.

Reply:
I don't believe "isInfantry" has anything to do with this matter, beside being the old name for "isLandTransportable". I'm certain that "isLandTransportable" didn't yet exist when "isInfantry" was non-deprecated and "isMechanized" was.

If "isMechanized" is merely an old name for "isInfantry", then "isInfantry" would be the intermediate rename between "isMechanized" and "isLandTransportable", but I believe this is not actually the case, as I recall that "isInfantry" used to be an option applying more generally (I think it might have been the reference used for paratroopers too, at some point, but I'm not sure about this) and I don't even know which one existed before, in TripleA, between "isMechanized" and "isInfantry". I vaguely recall that the application of "isInfantry" varied overtime, until being restricted to, then superseded by, what "isLandTransportable" is currently doing, but I don't have the details.

My suggestion is not caring about the history of "isInfantry". What is actually needed is to assure that the current behaviour of "isLandTransportable" is exactly the same as the pre-deprecation behaviour of "isMechanized", if it is really wanted to suggest such a change (instead of telling to delete "isMechanized").

Reply:
IIRC 'isInfantry' implied 'isMechanized' and a few other properties and that is why it was removed.

'isMechanized' is seemingly properly replaced by 'isLandTransportable'. For example, I found this check for it a unit can be loaded when the 'mechanized infantry' tech is discovered:

```
    if (getIsLandTransportable() && playerHasMechInf(player)) {
      tuples.add(Tuple.of("Can be Land Transported", ""));
    }
```

Reply:
I recall that "isInfantry" was renamed to "isLandTransportable" merely to have a clearer name. When that happened, TripleA was past the point in which I believe you completely removed "isMechanized" from the program (under my suggestion).

I believe what you quote may have nothing to do with the matter at hand. "MechInf" is referring to a technology a player may have, while "isMechanized" is a unit option. Unless proved by other means, I don't think that similar names assume relation (we have cases they do not, like "kamikaze"). Even if it is likely, in this case, that they are related, this still is not assuring that the non-deprecated behaviour of a "isMechanized" true unit would be the same as the current behaviour of a "isLandTransportable" true unit.

Reply:
I'm pretty sure 'isInfantry' was not renamed, but 'isMechanized' was. 'isinfantry' granted more options other than just land transportability.

Reply:
I'm pretty confident that the noted changes are correct. Could certainly be wrong, it's an overly complicated mess of a code-base (that does great things, but messy under the covers). If we find out otherwise we can fix it. In the meantime I suspect nobody is going to use this guide for quite some time and perhaps if they can bring it up. For practical purposes this guide is actually most useful to me right now for coding up how legacy options are to be ported forward.

Reply:
"isInfantry" was not renamed? I'm as sure as I can be that @ron-murhammer renamed "isInfantry" to "isLandTransportable" and I still believe this has nothing to do with this matter. If the "isInfantry" to "isLandTransportable" would have been never made, I think now we would be at the same point, just that "isLandTransportable" would be still called "isInfantry". Then, we would be just discussing whether or not to suggest changing "isMechanized" to "isInfantry", exactly like now we are discussing the same for "isMechanized" to "isLandTransportable".

If @ron-murhammer can confirm what I'm saying, otherwise someone would need to look at the code, I assume.

Reply:
No, isInfantry was not renamed as it was triggering multiple behaviors, including 'isLandTransportable'. "isInfantry' implies 'isLandTransportable', but not the other way around.

Reply:
I clearly remember that "isInfantry" was renamed to "isLandTransportable", while another option named "isInfantry" was added for compatibility. The "isInfantry" after the addition of "isLandTransportable" would do nothing but setting "isLandTransportable" to its same value.

Meaning that I believe the only reason why "isInfantry" still exists is to support old maps, just like if you would update them all changing "isInfantry" to "isLandTransportable".

Immediately before the addition of "isLandTransportable", I recall "isInfantry" was doing nothing more than what "isLandTransportable" currently does ("isInfantry" was probably doing other things further in the past, before 1.8.0.9).

>isInfantry was not renamed as it was triggering multiple behaviors, including 'isLandTransportable'.

I believe that, from the addition of "isLandTransportable" onwards, "isInfantry" does nothing but setting "isLandTransportable". So I'm not aware that now or at any point in time "isInfantry" has ever both set "isLandTransportable" and done anything else, at the same time.

>"isInfantry' implies 'isLandTransportable', but not the other way around.

As far as I know, this is true only because "isInfantry" does nothing but setting "isLandTransportable" to its same value. If "isInfantry" currently does anything else (making "isLandTransportable" a subset of "isInfantry"), this should be documented, I assume.

@ron-murhammer Since I recall you are behind the "isLandTransportable" addition, can you help clarifying the matter here?

Reply:
It could be the case we were in a partial state where 'isInfantry' was made to be a subset of its original functionality. I would be a bit surprised AFAIK the only mass migrations of maps was done when we went to 1.9.

Currently 'isInfantry' appears nowhere in the codebase except in the legacy property mapper.

Reply:
'isInfantry' is missing from the list actually. We are beyond the scope of this update @Cernelius . It's best practice to keep conversations in a PR to the content in the PR. If we want clarification on this, let's discuss in an issue so that others can find it later and not have to search random PRs for the convo.

Reply:
> 'isInfantry' is missing from the list actually. We are beyond the scope of this update @Cernelius . It's best practice to keep conversations in a PR to the content in the PR. If we want clarification on this, let's discuss in an issue so that others can find it later and not have to search random PRs for the convo.

This is exactly what I said:

>I don't believe "isInfantry" has anything to do with this matter, beside being the old name for "isLandTransportable". I'm certain that "isLandTransportable" didn't yet exist when "isInfantry" was non-deprecated and "isMechanized" was.

We should have really never even mentioned "isInfantry", as the matter here appears to be exclusively whether to suggest to delete "isMechanized" as an option or to change "isMechanized" to "isLandTransportable". I believe this is dependent exclusively on what these two options did or do. You added "isInfantry" to the matter, as an element to take into consideration.